### PR TITLE
Add multipart request

### PR DESCRIPTION
**Pull request by samwilson**
For: https://phabricator.wikimedia.org/T177266

### CODE REVIEW

File: src/MultipartRequest.php [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+namespace Mediawiki\Api;
+
+use Exception;
+
+/**
+ * A MultipartRequest is the same as a FluentRequest with additional support for setting request
+ * parameters (both normal parameters and headers) on multipart requests.
+ *
+ * @link http://docs.guzzlephp.org/en/stable/request-options.html#multipart
```
Comment (addshore) [inline src/MultipartRequest.php:11]: @since please! :D

**Comment by addshore** [comment]
Looks good, but needs a rebase and there is 1 comment.

**Comment by samwilson** [comment]
I've set it as `@since 2.4.0`, which is the next minor version. Is that correct?

**Comment by samwilson** [comment]
I've added some docs etc. See what you think now.

**Pull request merged**